service/dap/server_test.go

```diff
@@ -1893,6 +1884,121 @@ func TestEvaluateRequest(t *testing.T) {
 	})
 }
 
+func TestEvaluateRequestLongStr(t *testing.T) {
+	runTest(t, "largevariables", func(client *daptest.Client, fixture protest.Fixture) {
+		runDebugSessionWithBPs(t, client, "launch",
+			// Launch
+			func() {
+				client.LaunchRequest("exec", fixture.Path, !stopOnEntry)
+			},
+			// Breakpoints are set within the program
+			fixture.Source, []int{},
+			[]onBreakpoint{{
+				execute: func() {
+					client.StackTraceRequest(1, 0, 20)
+					stack := client.ExpectStackTraceResponse(t)
+					expectStackFrames(t, stack, "main.f", -1, 1000, 4, 4)
+
+					testCases := []struct {
+						expr         string
+						wantShort    string
+						wantLong     string
+						wantChildren bool
+					}{
+						{
+							expr:      "str",
+							wantShort: fmt.Sprintf(`"%v"`, strings.Repeat("a", 33)),
+							wantLong:  fmt.Sprintf(`"%v"`, strings.Repeat("a", 33)),
+						},
+						{
+							expr:      "longStr",
+							wantShort: fmt.Sprintf(`"%v...+36 more"`, strings.Repeat("a", 64)),
+							wantLong:  fmt.Sprintf(`"%v"`, strings.Repeat("a", 100)),
+						},
+						{
+							expr:      "veryLongStr",
+							wantShort: fmt.Sprintf(`"%v...+961 more"`, strings.Repeat("a", 64)),
+							wantLong:  fmt.Sprintf(`"%v...+1 more"`, strings.Repeat("a", 1024)),
+						},
+						{
+							expr:      "localStr",
+							wantShort: fmt.Sprintf(`"%v"`, strings.Repeat("a", 33)),
+							wantLong:  fmt.Sprintf(`"%v"`, strings.Repeat("a", 33)),
+						},
+						{
+							expr:      "localLongStr",
+							wantShort: fmt.Sprintf(`"%v...+36 more"`, strings.Repeat("a", 64)),
+							wantLong:  fmt.Sprintf(`"%v"`, strings.Repeat("a", 100)),
+						},
+						{
+							expr:      "localVeryLongStr",
+							wantShort: fmt.Sprintf(`"%v...+961 more"`, strings.Repeat("a", 64)),
+							wantLong:  fmt.Sprintf(`"%v...+1 more"`, strings.Repeat("a", 1024)),
+						},
+						{
+							expr:      "localStr[1:]",
+							wantShort: fmt.Sprintf(`"%v"`, strings.Repeat("a", 32)),
+							wantLong:  fmt.Sprintf(`"%v"`, strings.Repeat("a", 32)),
+						},
+						{
+							expr:      "localLongStr[1:]",
+							wantShort: fmt.Sprintf(`"%v...+35 more"`, strings.Repeat("a", 64)),
+							wantLong:  fmt.Sprintf(`"%v"`, strings.Repeat("a", 99)),
+						},
+						{
+							expr:      "localVeryLongStr[1:]",
+							wantShort: fmt.Sprintf(`"%v...+960 more"`, strings.Repeat("a", 64)),
+							wantLong:  fmt.Sprintf(`"%v"`, strings.Repeat("a", 1024)),
+						},
+						{
+							expr:      "string(bytes)",
+							wantShort: fmt.Sprintf(`"%v"`, strings.Repeat("a", 33)),
+							wantLong:  fmt.Sprintf(`"%v"`, strings.Repeat("a", 33)),
+						},
+						// EvalVariableInScope doesn't seem to use load config when
+						// evaluating []byte to string conversion. Probably not critical
+						// but figure out why.
```

Comment:
Add TODO?

Reply:
[]byte is not handled in this PR any more so the code is gone.